common/lib/xmodule/xmodule/modulestore/draft_and_published.py

```diff
@@ -134,6 +134,38 @@ def _flag_publish_event(self, course_key):
                 # We remove the branch, because publishing always means copying from draft to published
                 self.signal_handler.send("course_published", course_key=course_key.for_branch(None))
 
+    def remove_reference_if_moved(self, item_key, source_parent_location, user_id):
+        """
+        Removes moved block reference from children list of it's moved parent.
+
+        Arguments:
+            item_key (BlockUsageLocator)                    : Locator of item.
+            source_parent_location (BlockUsageLocator)      : Original parent block locator.
+            user_id (int)                                   : User id
+
+        Returns:
+           location                                         : Locator of item.
+        """
+        try:
+            item = self.get_item(item_key)
+        except ItemNotFoundError:
+            return
+        item_parent_location = unicode(item.parent.for_branch(None))
+        if item_parent_location and item_parent_location != unicode(source_parent_location):
+            # If an item parent is different than the current parent then it means it is moved.
```

Comment:
Change to:
` Item's parent is different than its new parent - so it has moved.`